src/mesh/voxel_cloud.rs

```diff
@@ -141,6 +142,169 @@ impl VoxelCloud {
         voxel_cloud
     }
 
+    /// Creates a new voxel cloud with arbitrary voxel dimensions from another
+    /// voxel cloud.
+    /// # Panics
+    /// Panics if any of the voxel dimensions is below or equal to zero.
+    pub fn from_voxel_cloud(
+        source_voxel_cloud: &VoxelCloud,
+        voxel_dimensions: &Vector3<f32>,
+    ) -> Option<Self> {
+        assert!(
+            voxel_dimensions.x > 0.0 && voxel_dimensions.y > 0.0 && voxel_dimensions.z > 0.0,
+            "One mor more voxel dimensions is below or equal to zero"
+        );
+
+        source_voxel_cloud
+            .mesh_volume_bounding_box_cartesian()
+            .map(|current_vc_bounding_box| {
+                // Make a bounding box of the source bounding box's mesh volume.
+                // This will be the volume to be scanned for any voxels.
+
+                // New voxel cloud that can encompass the source
+                // voxel cloud's mesh.
+                let mut target_voxel_cloud = VoxelCloud::from_cartesian_bounding_box(
+                    &current_vc_bounding_box,
+                    &voxel_dimensions,
+                );
+
+                for (one_dimensional, voxel) in target_voxel_cloud.voxel_map.iter_mut().enumerate()
+                {
+                    let cartesian_coordinate = one_dimensional_to_cartesian_coordinate(
+                        one_dimensional,
+                        &target_voxel_cloud.block_start,
+                        &target_voxel_cloud.block_dimensions,
+                        &target_voxel_cloud.voxel_dimensions,
+                    )
+                    .expect("Index out of bounds");
+
+                    // Set the new voxel state according to a sampled value from
+                    // the source voxel cloud.
+                    *voxel = source_voxel_cloud
+                        .voxel_at_cartesian_coords(&cartesian_coordinate)
+                        .unwrap_or(false);
+                }
+
+                target_voxel_cloud
+            })
+    }
+
+    /// Creates a new voxel cloud from another voxel cloud transformed (scaled,
+    /// rotated, moved) in a cartesian space.
+    /// # Panics
+    /// Panics if any of the voxel dimensions is below or equal to zero.
+    pub fn from_voxel_cloud_transformed(
+        source_voxel_cloud: &VoxelCloud,
+        voxel_dimension: f32,
+        cartesian_translation: &Vector3<f32>,
+        rotation_degrees: &(f32, f32, f32),
+        scale: &(f32, f32, f32),
```

Comment:
I know this might sound arbitrary, but let `scale`be `(f32, f32, f32)`. The type is quite small. It's more like we made an exception for the nalgebra types.

Or even better, let it be a `&Vector`. You construct one anyway later in the code.